mscore/startupWizard.cpp

```diff
@@ -0,0 +1,80 @@
+//=============================================================================
+//  MuseScore
+//  Music Composition & Notation
+//  Copyright (C) 2017 Werner Schweer and others
+//
+//  This program is free software; you can redistribute it and/or modify
+//  it under the terms of the GNU General Public License version 2.
+//
+//  This program is distributed in the hope that it will be useful,
+//  but WITHOUT ANY WARRANTY; without even the implied warranty of
+//  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+//  GNU General Public License for more details.
+//
+//  You should have received a copy of the GNU General Public License
+//  along with this program; if not, write to the Free Software
+//  Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
+//=============================================================================
+
+#include "startupWizard.h"
+#include "musescore.h"
+#include "libmscore/mscore.h"
+#include "shortcut.h"
+
+namespace Ms {
+QString StartupWizard::layoutInput()
+      {
+      QStringList layoutList;
+      QMap<QString, QString> layoutToShortcut = StartupWizard::getShortcutMap();
+      QList<QString> keyboardLayouts = layoutToShortcut.keys();
+      for (auto layout : keyboardLayouts) {
+            layoutList.append(layout);
+      }
+
+      QString labelMessage = QObject::tr("Enter your keyboard layout");
+      QString result = QInputDialog::getItem(new QWidget(), "Startup Wizard", labelMessage, layoutList);
+      return result;
+      }
+
+void StartupWizard::autoSelectShortcuts(QString keyboardLayout)
+      {
+      QMap<QString, QString> layoutToShortcut = getShortcutMap();
+      QString fileLocation;
+      try {
+            fileLocation = layoutToShortcut.value(keyboardLayout);
+      } catch (...) {
+            fileLocation = ":/data/shortcuts.xml";
+      }
+      Shortcut::loadFromNewFile(fileLocation);
+      }
+
+QMap<QString, QString> StartupWizard::getShortcutMap()
+      {
+      QMap<QString, QString> *layoutToShortcut = new QMap<QString, QString>();
```

Comment:
with `new` you are requesting memory. Where, when is it freed, when is that object deleted?